File: packages/typra-emitter/fixtures/integration/vector-adapters/typescript/vector-adapters.ts
// Reference adapters for the integration fixture's @vector suite. Copied into
// each TypeScript target's tests/ dir by validate-fixtures before the generated
// conformance suite runs. See fixtures/integration/vector-adapters.
type ReferenceAdapter = {
  invoke: (input: unknown) => unknown;
};

export const vectorAdapters: Record<string, ReferenceAdapter> = {
  "CanonicalEnginePort.authorize": {
    invoke: () => ({ approved: true }),
  },
  "CanonicalEnginePort.format": {
    invoke: (input) => (input as { messages: unknown }).messages,
  },
};

export const vectorWaivers: Record<string, string> = {};
export const vectorDoubles: Record<string, unknown> = {};

File: packages/typra-emitter/fixtures/integration/vector-adapters/python/vector_adapters.py
"""Reference adapters for the integration fixture's @vector suite.

Copied into each Python target's tests/ dir by validate-fixtures before the
generated conformance suite runs. See fixtures/integration/vector-adapters.
"""


def _authorize(_input, _context):
    return {"approved": True}


def _format(input, _context):
    return input["messages"]


VECTOR_ADAPTERS = {
    "CanonicalEnginePort.authorize": {"invoke": _authorize},
    "CanonicalEnginePort.format": {"invoke": _format},
}

VECTOR_WAIVERS = {}
VECTOR_DOUBLES = {}
